front/app/containers/SiteMap/index.tsx

```diff
@@ -117,41 +114,39 @@ const NavItem = styled.button`
   }
 `;
 
-interface DataProps {
-  projects: GetProjectsChildProps;
-  authUser: GetAuthUserChildProps;
-}
-
-interface Props extends DataProps {}
-
-const SiteMap = ({ projects, authUser }: Props) => {
+const SiteMap = () => {
   const proposalsEnabled = useFeatureFlag({ name: 'initiatives' });
+  const projects = useProjects({
+    publicationStatuses: ['draft', 'published', 'archived'],
+  });
   const loaded = projects !== undefined;
   const navBarItems = useNavbarItems();
   const localize = useLocalize();
   const pages = useCustomPages();
-
-  const scrollTo = (component) => (event: any) => {
-    // if the event is synthetic, it's a key event and we move focus
-    // https://github.com/facebook/react/issues/3907
-    if (event.detail === 0) {
-      component.current.focus();
-    }
-    scrollToComponent(component.current, {
-      align: 'top',
-      offset: -90,
-      duration: 300,
-    });
-  };
-
-  const homeSection = useRef(null);
-  const projectsSection = useRef(null);
-  const archivedSection = useRef(null);
-  const currentSection = useRef(null);
-  const draftSection = useRef(null);
-  const initiativesSection = useRef(null);
-  const userSpaceSection = useRef(null);
-  const customPagesSection = useRef(null);
+  const authUser = useAuthUser();
+
+  const scrollTo =
+    (component: RefObject<HTMLHeadingElement | null>) =>
+    (event: MouseEvent | KeyboardEvent) => {
```

Comment:
Not sure how I best type events. The TS [cheatsheet](https://github.com/typescript-cheatsheets/react#list-of-event-types) would say this is a synthetic event, but then I can't access event.details.

Reply:
I think it's ok, the comment makes it very clear.